### PR TITLE
jtbl: init at 1.6.0

### PR DESCRIPTION
**Pull request by elliott-farrall**
[JTBL](https://github.com/kellyjonbrazil/jtbl) is a CLI tool to convert JSON and JSON Lines to terminal, CSV, HTTP, and markdown tables.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/jt/jtbl/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "jtbl";
+  version = "1.6.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "kellyjonbrazil";
+    repo = pname;
```
Comment (sikmir) [inline pkgs/by-name/jt/jtbl/package.nix:14]:
```suggestion
    repo = "jtbl";
```

File: pkgs/by-name/jt/jtbl/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    rev = "v${version}";
+    hash = "sha256-ILQwUpjNueaYR5hxOWd5kZSPhVoFnnS2FcttyKSTPr8=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
+    setuptools
+    wheel
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
```
Comment (sikmir) [inline pkgs/by-name/jt/jtbl/package.nix:24]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/by-name/jt/jtbl/package.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "kellyjonbrazil";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-ILQwUpjNueaYR5hxOWd5kZSPhVoFnnS2FcttyKSTPr8=";
+  };
+
+  nativeBuildInputs = with python3Packages; [
```
Comment (sikmir) [inline pkgs/by-name/jt/jtbl/package.nix:19]:
```suggestion
  build-system = with python3Packages; [
```

File: pkgs/by-name/jt/jtbl/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  pythonImportsCheck = [ "jtbl" ];
+
+  meta = with lib; {
+    description = "CLI tool to convert JSON and JSON Lines to terminal, CSV, HTTP, and markdown tables";
+    homepage = "https://kellyjonbrazil.github.io/jtbl";
+    downloadPage = "https://github.com/kellyjonbrazil/jtbl/releases/tag/${src.rev}";
+    changelog = "https://github.com/kellyjonbrazil/jtbl/blob/${src.rev}/CHANGELOG";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ElliottSullingeFarrall ];
+    platforms = platforms.unix;
```
Comment (romner-set) [inline pkgs/by-name/jt/jtbl/package.nix:37]: Does this need to be set? Upstream seems to support all platforms.

File: pkgs/by-name/jt/jtbl/package.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  pythonImportsCheck = [ "jtbl" ];
+
+  meta = with lib; {
+    description = "CLI tool to convert JSON and JSON Lines to terminal, CSV, HTTP, and markdown tables";
+    homepage = "https://kellyjonbrazil.github.io/jtbl";
+    downloadPage = "https://github.com/kellyjonbrazil/jtbl/releases/tag/${src.rev}";
+    changelog = "https://github.com/kellyjonbrazil/jtbl/blob/${src.rev}/CHANGELOG";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ElliottSullingeFarrall ];
```
Comment (romner-set) [inline pkgs/by-name/jt/jtbl/package.nix:36]: Nested `with`s should not be used.

```suggestion
  meta = {
    description = "CLI tool to convert JSON and JSON Lines to terminal, CSV, HTTP, and markdown tables";
    homepage = "https://kellyjonbrazil.github.io/jtbl";
    downloadPage = "https://github.com/kellyjonbrazil/jtbl/releases/tag/${src.rev}";
    changelog = "https://github.com/kellyjonbrazil/jtbl/blob/${src.rev}/CHANGELOG";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ ElliottSullingeFarrall ];
```

**Review by maevii — CHANGES_REQUESTED** [review]
Please remove the [Apply suggestions from code review](https://github.com/NixOS/nixpkgs/pull/346699/commits/8930c4e6e5b4d2dce553ca9cb32b02fca5195b2c) commit and rewrite [jtbl: init at 1.6.0](https://github.com/NixOS/nixpkgs/pull/346699/commits/196727316f033ff7574f831f17fa4cc3d6643833) to include the changes instead.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**